### DISCUSSION TITLE
Following "Multiple Models with FastAPI" results in shadowing error

### DISCUSSION THREAD

**Discussion by ganondev** [fastapi/sqlmodel#702] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
class Character(SQLModel, table=True):
    id: Optional[UUID] = Field(primary_key=True, index=True)


class CharacterResponse(Character):
    id: UUID
```

### Description

From [Multiple Models](https://sqlmodel.tiangolo.com/tutorial/fastapi/multiple-models/#the-heroread-data-model)
> Now let's check the HeroRead model.
>
> This one just declares that the id field is required when reading a hero from the API, because a hero read from the API will come from  the database, and in the database it will always have an ID.
>
>
> ```
> class HeroBase(SQLModel):
>    name: str = Field(index=True)
>    secret_name: str
>    age: Optional[int] = Field(default=None, index=True)
>
>
> class Hero(HeroBase, table=True):
>    id: Optional[int] = Field(default=None, primary_key=True)
>
>
> class HeroCreate(HeroBase):
>    pass
>
>
> class HeroRead(HeroBase):
>    id: int
>
> ```

Actually trying this gets you:
```
NameError: Field name "id" shadows a BaseModel attribute; use a different field name with "alias='id'".
```
When trying to use `HeroRead` as a response schema for a fastapi endpoint.

### Operating System

Linux

### Operating System Details

Docker base on `python:3.12`

### SQLModel Version

sqlmodel==0.0.11
pydantic is at 1.10.13, if it matters. I don't specify this, it is auto resolved for sqlmodel and fastapi.

### Python Version

3.12.0

### Additional Context

Posting this here instead of fastapi or pydantic because this seems like an intentional feature of those two, but conflicts with the aforementioned documentation.

**Comment by YuriiMotov** [comment]
I think this has been resolved. Can't reproduce in current version
